src/compile/legend.ts

```diff
@@ -6,13 +6,16 @@ import {AREA, BAR, TICK, TEXT, LINE, POINT, CIRCLE, SQUARE} from '../mark';
 import {extend, keys} from '../util';
 import {Model} from './Model';
 import {applyMarkConfig, FILL_STROKE_CONFIG, formatMixins as utilFormatMixins} from './util';
+import {ORDINAL} from '../type';
+import {INVERSE_RANK, COLOR_LABEL} from './scale';
 
 export function compileLegends(model: Model) {
   var defs = [];
 
   if (model.has(COLOR) && model.fieldDef(COLOR).legend) {
+    const fieldDef = model.fieldDef(COLOR);
     defs.push(compileLegend(model, COLOR, {
-      fill: model.scaleName(COLOR)
+      fill: (fieldDef.type === ORDINAL || fieldDef.bin) ? INVERSE_RANK : model.scaleName(COLOR)
```

Comment:
I assume that since temporal uses time scale, this will be correct for them.  

(If so, I guess this will still be incorrect for month, day, hour which has ordinal scale and thus need the same hack?)